app/services/agents/destroy_service.rb

```diff
@@ -0,0 +1,28 @@
+class Agents::DestroyService
+  pattr_initialize [:account!, :user!]
+
+  def perform
```

Comment:
should we wrap this up in a transaction to ensure we don't half deleted states ?

Reply:
Makes sense, will change it tomorrow.